.github/workflows/release_asset.yml

```diff
@@ -0,0 +1,51 @@
+name: Release asset bot
+
+on:
+  workflow_dispatch:
+  release:
+    types: [published]
+jobs:
+  build:
+    name: Build release asset
+    continue-on-error: true
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - ubuntu-latest
+          - ubuntu-18.04
+          - macos-latest
+          - windows-latest
+    runs-on: ${{matrix.platform}}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: install rust toolchain
+        uses: actions-rs/toolchain@v1
+      - name: build for ${{ matrix.platform }}
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --release
+
+      - name: build move release asset
+        run: bash ./scripts/release.sh ${{ matrix.platform }}
+
+      - name: upload artifact asset
+        uses: actions/upload-artifact@v2
+        if: ${{ github.event_name != 'release'}}
+        with:
+          name: move-${{ matrix.platform }}.zip
+          path: ./move-${{ matrix.platform }}.zip
+
+      - name: upload move release asset
+        if: ${{ github.event_name == 'release'}}
+        id: upload-release-asset
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ github.event.release.upload_url }}
+          asset_path: ./move-${{ matrix.platform }}.zip
+          asset_name: move-${{ matrix.platform }}.zip
+          asset_content_type: application/zip
```

Comment:
Would you elaborate, maybe even add a comment to the file, about the purpose of each of these different jobs?

My best guess is that one is for when a "release" even occurs (by creating a new tag) and the other is for manually triggering the job. Is that correct?

Reply:
ok, i'll write a code comment